packages/core/src/utils/shadowComparer.ts

```diff
@@ -0,0 +1,29 @@
+// shadow equality
+export const shadowComparer = <T>(
+  prev: T,
+  next: T,
+): boolean => {
+  const isDifferentTypes = typeof prev !== typeof next;
+
+  if (isDifferentTypes) {
+    return false;
+  }
+
+  const notReferenceTypes = !(prev instanceof Object) || !(next instanceof Object);
+  const isFunctionTypes = typeof prev === 'function' && typeof next === 'function';
+
+  if (notReferenceTypes || isFunctionTypes) {
+    return prev === next;
+  }
+
+  const prevObj = prev as { [index: string]: unknown };
+  const nextObj = next as { [index: string]: unknown };
```

Comment:
I think prev/next maybe confusing - it's not necessary that one follows another in some sequence

```suggestion
  const a = prev as { [index: string]: unknown };
  const b = next as { [index: string]: unknown };
```
Maybe `objA`, `objB`, or smth like.

Reply:
Let's name it ```firstValue``` and ```secondValue```